core/templates/domain/platform_feature/feature-status-summary-object.factory.ts

```diff
@@ -19,10 +19,40 @@
 import { Injectable } from '@angular/core';
 import { downgradeInjectable } from '@angular/upgrade/static';
 
+/**
+ * Names of all feature flags should be defined here, with format:
+ * FeatureName = 'feature_name', where the LHS is the feature name in
+ * PascalCase, and the RHS is in snake_case, which is the naming convention
+ * of features in the backend.
+ */
+export enum FeatureNames {
+  DummyFeature = 'dummy_feature',
+}
```

Comment:
Added tests for this in `core/platform_feature_list_test`, that reads and parse the ts file to check the existence of feature names.
This solution seems suboptimal, I considered defining feature names in `constants.ts` or `feconf.py` but they all results to lost of member hinting as they are not ts files...

Reply:
Why would we lose typing if in constants.ts? Can we not define an enum in there & cross-validate the list with a Python test?

Not keen on the existing solution needing to parse a frontend file. I don't think we do that elsewhere, and the recommended pattern for sharing constants between the frontend & backend is constants.ts.

Reply:
I tried to define enum in `constants.ts` but there are some issues:

* inside `constants.ts` it uses `exports = {}` which makes it impossible to have another exported entry with `export enum {}`
* it's possible though to change it to `exports default {}` so that we can have other export entries, but this affects the whole base, tens of files needs to be changed and I'm not sure if simply changing the reference will have any issues in some js environment
  * even if I did so, and I created an enum for the names, I cannot add all the names as a list in the default exported json (to make it visible for the python end), because it requires runtime computation which makes it a ts/js file instead of a json, then in py it cannot be simply parsed.
* If I chose to simply add a name list as in the big constant object inside `constants.ts`. then I am not able to convert the list to enum because the list is considered to be a runtime variable, not a static constant.

It might still be possible to do so, but I think the work needed is not trivial...

Reply:
Okay, thanks for digging into it. Let's keep the current solution, and we can iterate on this in the future if the current solution actually introduces issues. The backend tests to keep things in sync are a good idea.